hack/verify-golint.sh

```diff
@@ -18,7 +18,11 @@ set -o errexit
 set -o pipefail
 
 # golangci-lint binary path
+# Check if GOBIN is set, if not use GOPATH/bin
 golangci_lint_binary="$(go env GOPATH)/bin/golangci-lint"
+if [ -n "$(go env GOBIN)" ]; then
+    golangci_lint_binary="$(go env GOBIN)/golangci-lint"
```

Comment:
not sure if it is a good approach, by default the go_path structure should have bin, src and pkg directories.
GOBIN points to the golang binaries itself.

Reply:
By default, the GOPATH structure does include bin, src, and pkg directories, which is correct. However, GOBIN does not point to the Go binary itself (e.g., the go command). Instead, it specifies the directory where binaries installed via go install are placed.

If GOBIN is set, it takes precedence over the default GOPATH/bin for storing installed binaries. This behavior is explicitly documented in the Go documentation, and it allows users to customize the installation path for their binaries while keeping the Go toolchain itself separate. Using GOBIN in this way is a standard and recommended approach when a custom binary path is needed.

Reply:
by `go binary iteself`, I meant the place where binaries are placed when you run `go install`, which is by default gobin :)
sry for the confusion.